src/org/omegat/core/data/ImportFromAutoTMX.java

```diff
@@ -169,20 +169,14 @@ private boolean altTranslationMatches(PrepareTMXEntry entry, EntryKey key) {
                 path = p.getValue();
             }
         }
-        return key.equals(new EntryKey(file, entry.source, id, prev, next, path));
+        return key.equals(new EntryKey(file, entry.getSourceText(), id, prev, next, path));
     }
 
-    private void setTranslation(SourceTextEntry entry, PrepareTMXEntry trans, boolean defaultTranslation,
+    private void setTranslation(SourceTextEntry entry, ITMXEntry trans, boolean defaultTranslation,
             TMXEntry.ExternalLinked externalLinked) {
-        if (StringUtil.isEmpty(trans.note)) {
-            trans.note = null;
-        }
-
-        trans.source = entry.getSrcText();
-
         TMXEntry newTrEntry;
 
-        if (trans.translation == null && trans.note == null) {
+        if (trans.getTranslationText() == null && (trans.getNote() == null || trans.getNote().length() == 0)) {
```

Comment:
I would be OK with a `hasNote()` method that computes `note != null && !note.isEmpty()`

Reply:
OK, indeed the method was in TMXEntry, so I have moved it to the interface so that we are sure to have a consistent behaviour everywhere we would need it.
(in particular this line was not correct since it seems that OmegaT considers that empty note is different from no note at all, so now it is corrected)